content/articles/image-preprocessing-in-python/index.md

```diff
@@ -0,0 +1,293 @@
+
+```python
+# Importing and Loading the data into data frame
+
+dataset_path = '/content/drive/MyDrive/Animals'
+class_names = ['Cheetah', 'Jaguar', 'Leopard', 'Lion','Tiger']
+
+# apply glob module to retrieve files/pathnames  
+
+animal_path = os.path.join(dataset_path, class_names[1], '*')
+animal_path = glob.glob(animal_path)
+```
+
+Now that we have already loaded our dataset onto our workspace, the next step would be to visualize the data.
+
+In order to verify if our data is properly loaded, we shall try accessing an image file from the dataset:
+
+```python
+# accessing an image file from the dataset classes
+image = io.imread(animal_path[4])  
+
+# plotting the original image
+i, (im1) = plt.subplots(1)
+i.set_figwidth(15)
+im1.imshow(image)
+```
+
+By executing the above code block, we shall randomly print an image from the dataset.
+
+![plotted image](/engineering-education/image-preprocessing-in-python/plotted_image.jpg)
+
+### Data preprocessing
+
+Now that we're done with importing libraries and data, we shall go into data preprocessing. Since images exists in different formats, i.e natural, fake, grayscale etc., we need to take in these considerations and standardize them before feeding them into a neural network.
+
+In this tutorial, we shall look at the following image preprocessing techniques:
+
+1. Grayscale conversion
+2. Normalisation
+3. Data Augmentation
+4. Standardizing images
+
+Before we start exploring the preprocessing techniques, let's first explore the RGB channels of our original image;
+
+```python
+# plotting the original image and the RGB channels  
+
+i, (im1, im2, im3, im4) = plt.subplots(1, 4, sharey=True)
+i.set_figwidth(20) 
+
+im1.imshow(image)  #Original image
+im2.imshow(image[:, : , 0]) #Red
+im3.imshow(image[:, : , 1]) #Green
+im4.imshow(image[:, : , 2]) #Blue
+i.suptitle('Original & RGB image channels')
+```
+
+Output
+![rgb](/engineering-education/image-preprocessing-in-python/rgb.jpg)
+
+#### Grayscale conversion
+
+Grayscale is simply converting images from colored to black and white. It is normally used to reduce computation complexity in machine learning algorithms. Since most pictures don't need color to be recognized, it is wise to use grayscale which reduces the number of pixels in an image hence reducing the computations required.
+
+Execute the code below to convert the original image to grayscale:
+
+```python
+gray_image = skimage.color.rgb2gray(image)
+plt.imshow(gray_image, cmap = 'gray')
+```
+
+Output
+![gray_scale](/engineering-education/image-preprocessing-in-python/gray_scale.jpg)
+
+>**NB**: Converting images to grayscale might not always be practical in solving some problems. A few examples where it would impractical to use grayscale include: _traffic lights_, _healthcare diagnosis_, _autonomous vehicles_, _agriculture_, etc. The best way to know whether to use it or not depends on your human visual ability to be able to identify an object without colour.
+
+#### Normalization
```

Comment:
Explain why normalization works! Why would rescaling [0,255] to [-1,1] or [0,1] help? How would it help? Do this for all the techniques mentioned.